src/ui/settings/keyboard-settings-ui-handler.ts

```diff
@@ -24,81 +21,102 @@ import i18next from "i18next";
-    deleteText.setOrigin(0, 0.15);
-    deleteText.setPositionRelative(iconDelete, -deleteText.displayWidth - 2, 0);
-
-    this.settingsContainer.add(iconDelete);
-    this.settingsContainer.add(deleteText);
-
-    // Map the 'noKeyboard' layout options for easy access.
-    this.layout["noKeyboard"].optionsContainer = optionsContainer;
-    this.layout["noKeyboard"].label = label;
-
-    // Listen to the home and delete key presses
-    this.deleteKey = globalScene.input.keyboard?.addKey(Phaser.Input.Keyboard.KeyCodes.DELETE);
-    this.homeKey = globalScene.input.keyboard?.addKey(Phaser.Input.Keyboard.KeyCodes.HOME);
-    this.deleteKey?.on("up", this.deleteBinding, this);
-    this.homeKey?.on("up", this.resetBindings, this);
+
+    // Add instructions for deleting a mapping
+    this.addInstructionText("BUTTON_DELETE", i18next.t("settings:delete"));
 
     eventBus.on("keyboard/init", this.updateChosenKeyboardDisplay, this);
   }
 
   protected override tearDown(): void {
-    this.deleteKey?.off("up", this.deleteBinding, this);
-    this.homeKey?.off("up", this.resetBindings, this);
-
     eventBus.off("keyboard/init", this.updateChosenKeyboardDisplay, this);
 
     super.tearDown();
   }
 
+  public override show(): boolean {
+    if (super.show()) {
+      // If a keyboard is connected but the view hasn't been initialized for it, do it
+      if (this.noDeviceText.visible && globalScene.inputController.getActiveConfig(this.device)) {
+        this.updateChosenKeyboardDisplay();
+      }
+      // Listen to the "delete" key presses to clear an existing mapping
+      this.deleteKey = globalScene.input.keyboard?.addKey(Phaser.Input.Keyboard.KeyCodes.DELETE);
+      this.deleteKey?.on("up", this.deleteBinding, this);
+      return true;
+    }
+    return false;
+  }
+
+  protected override clear(): void {
+    this.deleteKey?.off("up", this.deleteBinding, this);
+
+    super.clear();
+  }
+
   /**
-   * Handle the home key press event: reset mappings for the current device
+   * Update the display of the chosen keyboard layout.
    */
-  private resetBindings(): void {
-    if (![UiMode.SETTINGS_KEYBOARD, UiMode.SETTINGS_GAMEPAD].includes(globalScene.ui.getMode())) {
-      return;
+  private updateChosenKeyboardDisplay(): void {
+    this.noDeviceText.setVisible(false);
+    this.optionsContainer.setVisible(true);
+
+    // Update any bindings that might have changed since the last update.
+    this.initBindings();
+    this.updateInstructionIcons();
+
+    // Iterate over the keys in the settingDevice enumeration.
+    for (const [index, value] of Object.values(SettingKeyboard).entries()) {
+      if (value === "layout") {
+        // Update the text of the first option label under the current setting to the name of the chosen layout
+        this.updateOptionValueLabel(
+          index,
+          0,
+          truncateString(enumValueToKey(KeyboardLayout, settings.keyboard.layout ?? KeyboardLayout.QWERTY), 25),
+        );
```

Comment:
```suggestion
        const layoutLabel = truncateString(enumValueToKey(KeyboardLayout, settings.keyboard.layout ?? KeyboardLayout.QWERTY), 25);
        this.updateOptionValueLabel(index, 0, layoutLabel);
```

Reply:
actually, updating this to 
```suggestion
        const layoutLabel = truncateString(enumValueToKey(KeyboardLayout, settings.keyboard.layout), 25);
        this.updateOptionValueLabel(index, 0, layoutLabel);
```
because the ?? shouldn't be needed